pkg/util/debug_logging.go

```diff
@@ -127,8 +135,40 @@ func (r *reducedRedudancyLogHook) Levels() []logrus.Level {
 // Private - Nil Logging Formatter
 // -----------------------------------------------------------------------------
 
-type nilFormatter struct{}
+// nilFormatter is a logrus.Formatter that allowed "flipping a switch" that will
+// cause the logger to stop emitting output.
+type nilFormatter struct {
+	embeddedFormatter logrus.Formatter
+	drop              bool
+	lock              *sync.RWMutex
+}
+
+func newNilFormatter(embeddedFormatter logrus.Formatter) *nilFormatter {
+	return &nilFormatter{
+		embeddedFormatter: embeddedFormatter,
+		lock:              &sync.RWMutex{},
+	}
+}
 
 func (n *nilFormatter) Format(entry *logrus.Entry) ([]byte, error) {
-	return nil, nil
+	n.lock.RLock()
+	defer n.lock.RUnlock()
```

Comment:
Nice, prevent concurrent write. 
query: currently, does every module use this log components ?

Reply:
Currently only the manager uses it, and only if it detects from the environment that the system is under test.